pkg/ironic/ironic.go

```diff
@@ -126,28 +126,30 @@ func ensureIronicService(cctx ControllerContext, ironic *metal3api.Ironic) (Stat
 	return getServiceStatus(service)
 }
 
-func removeIronicDaemonSet(cctx ControllerContext, ironic *metal3api.Ironic) error {
+func removeIronicDaemonSet(cctx ControllerContext, ironic *metal3iov1alpha1.Ironic) error {
 	err := cctx.KubeClient.AppsV1().DaemonSets(ironic.Namespace).
 		Delete(context.Background(), ironicDeploymentName(ironic), metav1.DeleteOptions{})
 	return client.IgnoreNotFound(err)
 }
 
-func removeIronicDeployment(cctx ControllerContext, ironic *metal3api.Ironic) error {
+func removeIronicDeployment(cctx ControllerContext, ironic *metal3iov1alpha1.Ironic) error {
 	err := cctx.KubeClient.AppsV1().Deployments(ironic.Namespace).
 		Delete(context.Background(), ironicDeploymentName(ironic), metav1.DeleteOptions{})
 	return client.IgnoreNotFound(err)
 }
 
 // EnsureIronic deploys Ironic either as a Deployment or as a DaemonSet.
-func EnsureIronic(cctx ControllerContext, ironic *metal3api.Ironic, db *metal3api.IronicDatabase, apiSecret *corev1.Secret) (status Status, err error) {
+func EnsureIronic(cctx ControllerContext, ironic *metal3iov1alpha1.Ironic, db *metal3iov1alpha1.IronicDatabase, apiSecret *corev1.Secret) (status Status, err error) {
 	if db != nil && !isReady(db.Status.Conditions) {
 		cctx.Logger.Info("database is not ready yet")
 		return
 	}
 
-	if validationErr := metal3api.ValidateIronic(&ironic.Spec, nil); validationErr != nil {
+	validationErr := metal3iov1alpha1.ValidateIronic(&ironic.Spec, nil)
+	if validationErr != nil {
 		status = Status{Fatal: validationErr}
-		return
+		//lint:ignore nilerr reason
+		return // ignore
```

Comment:
```suggestion
		return //nolint:nilerr
```
That should ignore it in more concise and consistent way.

Though this looks like it would also warrant proper fix. Returning nothing instead of two values is sketchy. It can be fixed in later PRs as well.

Reply:
yes, honestly I forgot to change it with the one liner!
And I agree this should be fixed and not ignored, but it's not doing too much harm AFAICS
I'm planning to take care of the rest of the stuff we ignored commenting out the linters, I'll add this one to the list